packages/bff/src/services/agreementService.ts

```diff
@@ -57,3 +153,221 @@ export const getLatestAgreement = async (
     }
   })[0];
 };
+
+function isUpgradable(
+  descriptor: catalogApi.EServiceDescriptor,
+  agreement: agreementApi.Agreement,
+  descriptors: catalogApi.EServiceDescriptor[]
+): boolean {
+  return descriptors
+    .filter((d) => Number(d.version) > Number(descriptor.version))
+    .some(
+      (d) =>
+        (d.state === "PUBLISHED" || d.state === "SUSPENDED") &&
+        (agreement.state === "ACTIVE" || agreement.state === "SUSPENDED")
+    );
+}
+
+async function enhanceAgreementOverview(
+  agreement: agreementApi.Agreement,
+  clients: PagoPAInteropBeClients,
+  ctx: WithLogger<BffAppContext>
+): Promise<bffApi.AgreementListEntry> {
+  const { consumer, producer, eservice } = await getConsumerProducerEservice(
+    agreement,
+    clients,
+    ctx
+  );
+
+  const currentDescriptor = getCurrentDescriptor(eservice, agreement);
+
+  return {
+    id: agreement.id,
+    state: agreement.state,
+    consumer: {
+      id: consumer.id,
+      name: consumer.name,
+      kind: consumer.kind,
+    },
+    eservice: toCompactEservice(eservice, producer),
+    descriptor: toCompactDescriptor(currentDescriptor),
+    canBeUpgraded: isUpgradable(
+      currentDescriptor,
+      agreement,
+      eservice.descriptors
+    ),
+    suspendedByConsumer: agreement.suspendedByConsumer,
+    suspendedByProducer: agreement.suspendedByProducer,
+    suspendedByPlatform: agreement.suspendedByPlatform,
+  };
+}
+
+export async function enhanceAgreementDetailed(
+  agreement: agreementApi.Agreement,
+  clients: PagoPAInteropBeClients,
+  ctx: WithLogger<BffAppContext>
+): Promise<bffApi.Agreement> {
+  const { consumer, producer, eservice } = await getConsumerProducerEservice(
+    agreement,
+    clients,
+    ctx
+  );
+
+  const currentDescriptior = getCurrentDescriptor(eservice, agreement);
+
+  const activeDescriptor = eservice.descriptors
+    .toSorted((a, b) => Number(a.version) - Number(b.version))
+    .at(-1);
+  const activeDescriptorAttributes = activeDescriptor
+    ? descriptorAttributesIds(activeDescriptor)
+    : [];
+  const allAttributesIds = Array.from(
+    new Set([...activeDescriptorAttributes, ...tenantAttributesIds(consumer)])
+  );
```

Comment:
here we should have a utility like `toSetToArray` that could be used (maybe isn't already in the main branch https://github.com/pagopa/interop-be-monorepo/pull/823/files#diff-f88ff5f470c7f182a6d7e7da307feb2e06453438f12d39c098dcf96c73f1635c)